command/meta_backend.go

```diff
@@ -562,6 +562,76 @@ func (m *Meta) backendFromConfig(opts *BackendOpts) (backend.Backend, tfdiags.Di
 	}
 }
 
+// backendFromState retunrs the initialized (not configured) backend directly
+// from the state. This should be used only when a user runs `terraform init
+// -backend=false`. This function returns a local backend if there is no state
+// or no backend configured.
+func (m *Meta) backendFromState() (backend.Backend, tfdiags.Diagnostics) {
+	var diags tfdiags.Diagnostics
+	// Get the path to where we store a local cache of backend configuration
+	// if we're using a remote backend. This may not yet exist which means
+	// we haven't used a non-local backend before. That is okay.
+	statePath := filepath.Join(m.DataDir(), DefaultStateFilename)
```

Comment:
Are there scenarios where this can fail before the `sMgr` creation?

Reply:
There isn't really anything to fail here; `m.DataDir` and `filepath.Join` both return strings and m.DataDir returns a default value if the data dir isn't set. 

If there was a an inaccurate result from either (not pointing to the actual state), the error would appear two lines down when terraform tries to refresh.